File: BambooBank-Spring/src/main/java/BambooBank_Angular_Spring/BambooBank_Spring/repositories/EmployeeRepository.java
package BambooBank_Angular_Spring.BambooBank_Spring.repositories;


import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import BambooBank_Angular_Spring.BambooBank_Spring.pojos.Employee;


@Repository
public interface EmployeeRepository extends JpaRepository<Employee, Long> {

    List<Employee> findByFirstName(String firstName);

    List<Employee> findByLastName(String lastName);

    List<Employee> findByEmail(String email);

    List<Employee> findByAddress(String address);

    List<Employee> findBySalary(BigDecimal salary);

    List<Employee> findByHireDate(LocalDate hireDate);
}

File: BambooBank-Spring-Kotlin/src/main/kotlin/bamboobank_spring/bamboobankspringkotlin/repositories/EmployeeRepository.kt
package bamboobank_spring.bamboobankspringkotlin.repositories

import bamboobank_spring.bamboobankspringkotlin.pojos.Employee
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.math.BigDecimal
import java.time.LocalDate

@Repository
interface EmployeeRepository : JpaRepository<Employee, Long> {

    fun findByFirstName(firstName: String): List<Employee>

    fun findByLastName(lastName: String): List<Employee>

    fun findByEmail(email: String): List<Employee>

    fun findByAddress(address: String): List<Employee>

    fun findBySalary(salary: BigDecimal): List<Employee>

    fun findByHireDate(hireDate: LocalDate): List<Employee>
}